src/data/complex/basic.lean

```diff
@@ -461,10 +461,62 @@ by simp only [sub_conj, of_real_mul, of_real_one, of_real_bit0, mul_right_comm,
 
 /-! ### Absolute value -/
 
-/-- The complex absolute value function, defined as the square root of the norm squared. -/
-@[pp_nodot] noncomputable def abs (z : ℂ) : ℝ := (norm_sq z).sqrt
+namespace abs_theory
+-- We develop enough theory to bundle `abs` into an `absolute_value` before making things public;
+-- this is so there's not two versions of it hanging around.
 
-local notation `abs'` := has_abs.abs
+local notation (name := abs) `abs` z := ((norm_sq z).sqrt)
+
+private lemma mul_self_abs (z : ℂ) : (abs z) * (abs z) = norm_sq z :=
+real.mul_self_sqrt (norm_sq_nonneg _)
+
+private lemma abs_nonneg' (z : ℂ) : 0 ≤ abs z :=
+real.sqrt_nonneg _
+
+lemma abs_conj (z : ℂ) : (abs (conj z)) = abs z :=
+by simp
+
+private lemma abs_re_le_abs (z : ℂ) : |z.re| ≤ abs z :=
+begin
+  rw [mul_self_le_mul_self_iff (abs_nonneg z.re) (abs_nonneg' _),
+       abs_mul_abs_self, mul_self_abs],
+  apply re_sq_le_norm_sq
+end
+
+private lemma re_le_abs (z : ℂ) : z.re ≤ abs z :=
+(abs_le.1 (abs_re_le_abs _)).2
+
+private lemma abs_mul (z w : ℂ) : (abs (z * w)) = (abs z) * abs w :=
+by rw [norm_sq_mul, real.sqrt_mul (norm_sq_nonneg _)]
+
+private lemma abs_add (z w : ℂ) : (abs (z + w)) ≤ (abs z) + abs w :=
+(mul_self_le_mul_self_iff (abs_nonneg' (z + w))
+  (add_nonneg (abs_nonneg' z) (abs_nonneg' w))).2 $
+begin
+  rw [mul_self_abs, add_mul_self_eq, mul_self_abs, mul_self_abs, add_right_comm, norm_sq_add,
+      add_le_add_iff_left, mul_assoc, mul_le_mul_left (@zero_lt_two ℝ _ _),
+      ←real.sqrt_mul $ norm_sq_nonneg z, ←norm_sq_conj w, ←map_mul],
+  exact re_le_abs (z * conj w)
+end
+
+noncomputable def _root_.complex.abs : absolute_value ℂ ℝ :=
```

Comment:
Should we bundle it as an `absolute_value` or migrate to `has_norm.norm`?

Reply:
I'm happy to do either; this made the most sense to me as I'm currently getting rid of `is_absolute_value`; I also see that the current instance for this is in `analysis/complex/basic`, which needs `complex/determinant` so there would need to be a more major re-organising of files needed for this.